lighthouse-core/audits/byte-efficiency/byte-efficiency-audit.js

```diff
@@ -89,47 +91,85 @@ class UnusedBytes extends Audit {
+    const simulationBeforeChanges = new LoadSimulator(graph).simulate();
+    const resultsByUrl = new Map();
+    for (const result of results) {
+      resultsByUrl.set(result.url, result);
     }
 
+    // Update all the transfer sizes to reflect implementing our recommendations
+    graph.traverse(node => {
+      if (node.type !== 'network') return;
+      if (!resultsByUrl.has(node.record.url)) return;
+      const original = node.record.transferSize;
+      const wastedBytes = resultsByUrl.get(node.record.url).wastedBytes;
+      // cloning NetworkRequest objects is difficult, so just stash the original transfer size
+      node.record._originalTransferSize = original;
+      node.record._transferSize = Math.max(original - wastedBytes, 0);
+    });
+
+    const simulationAfterChanges = new LoadSimulator(graph).simulate();
+    // Restore the original transfer size after we've done our simulation
+    graph.traverse(node => {
+      if (node.type !== 'network') return;
+      if (!node.record._originalTransferSize) return;
+      node.record._transferSize = node.record._originalTransferSize;
+    });
+
+    const savingsOnTTI = Math.max(
+      PredictivePerf.getLastLongTaskEndTime(simulationBeforeChanges.nodeTiming) -
+        PredictivePerf.getLastLongTaskEndTime(simulationAfterChanges.nodeTiming),
+      0
+    );
+
+    // Round waste to nearest 10ms
+    return Math.round(savingsOnTTI / 10) * 10;
+  }
+
+  /**
+   * @param {!Audit.HeadingsResult} result
+   * @param {!Node} graph
+   * @return {!AuditResult}
+   */
+  static createAuditResult(result, graph) {
     const debugString = result.debugString;
     const results = result.results
-        .map(item => {
-          const wastedPercent = 100 * item.wastedBytes / item.totalBytes;
-          item.wastedKb = this.bytesToKbString(item.wastedBytes);
-          item.wastedMs = this.bytesToMsString(item.wastedBytes, networkThroughput);
-          item.totalKb = this.bytesToKbString(item.totalBytes);
-          item.totalMs = this.bytesToMsString(item.totalBytes, networkThroughput);
-          item.potentialSavings = this.toSavingsString(item.wastedBytes, wastedPercent);
-          return item;
-        })
-        .sort((itemA, itemB) => itemB.wastedBytes - itemA.wastedBytes);
+      .map(item => {
+        const wastedPercent = 100 * item.wastedBytes / item.totalBytes;
+        item.wastedKb = this.bytesToKbString(item.wastedBytes);
+        item.totalKb = this.bytesToKbString(item.totalBytes);
+        item.potentialSavings = this.toSavingsString(item.wastedBytes, wastedPercent);
+        return item;
+      })
+      .sort((itemA, itemB) => itemB.wastedBytes - itemA.wastedBytes);
 
     const wastedBytes = results.reduce((sum, item) => sum + item.wastedBytes, 0);
     const wastedKb = Math.round(wastedBytes / KB_IN_BYTES);
-    const wastedMs = Math.round(wastedBytes / networkThroughput * 100) * 10;
+    const wastedMs = this.computeWasteWithTTIGraph(results, graph);
```

Comment:
`results` don't have `url`s? (needed by `computeWasteWithTTIGraph`)

Reply:
but they do? :)

Reply:
Ah, I also missed that it was `result.results.map(...)`